src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.LocalFunctions.cs

```diff
@@ -163,9 +165,15 @@ public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatemen
             {
                 this.State = pending.State;
                 BoundNode branch = pending.Branch;
-                LeaveParameters(localFuncSymbol.Parameters, branch?.Syntax,
-                                branch?.WasCompilerGenerated == true
-                                    ? location : null);
+
+                // No need to check the parameters if we dont know the branch
+                if (branch != null)
```

Comment:
How likelly is it that the SyntaxNode for branch and localFunction could be null?